### ISSUE TITLE
check handling of multiple registrations by xform(image)

### ISSUE THREAD

**Issue by jefferis** [jefferis/nat#199]
need to ensure that this is consistent with xform(points) and that it seems sensible to most people – i.e. reflects expectation rather than implementation.

**Comment by jefferis** [comment]
@ajdm this is the one I was talking about. Here is an implementation of nat.templatebrain's that makes the necessary changes

```r
# reverses order of registations and removes  --inverse switch
# needed by streamxform but not reformatx.
xform_brain_image<-function (x, sample, reference, via = NULL, ...)
{
    regs <- rev(bridging_sequence(reference = reference, sample = sample,
        via = via, checkboth = T, mustWork = T))
    directions <- sapply(regs, function(reg) ifelse(!isTRUE(attr(reg, 
        "swapped")), "forward", "inverse"))
    nat::xform(x, reg = as.character(regs), direction = directions,
        ...)
}
``

Note the `rev(` and the `!isTRUE(`.  We would ideally like a way to do this somewhere in the xform/xformimages/xformpoints chain.

**Comment by jefferis** [comment]
After further thought:

1. all `xform.*` functions should expect registrations to be listed proceeding from sample to reference
2. this includes `nat.templatebrains::xform_brain` and `xformimages`
3. `xformimages.cmtkreg` should reverse the order of registrations as they are passed to `cmtk.reformatx` (which is a straightforward wrapper for CMTK reformatx)
4. `xform_brain` then does not need to worry about this ordering (but does need to know about imagedata for the purpose of checking cmtk registrations in both directions)

**Comment by jefferis** [comment]
This is now fixed, the key commit being 75b42e7daf1682a0acc96dcedd445408d9985fd2 as well as work related to https://github.com/jefferislab/nat.templatebrains/issues/11
